### PR TITLE
adopt ESM in tsconfig

### LINKED ISSUE #9184: adopt bundler moduleResolution in tsconfig
## What is the Problem Being Solved?

The Endo bundler uses a modern resolver that understand package "exports" map and disallows path-based imports like `/dist`.

Typescript also does in the Node16/NodeNext mode, but we use the default, Node10.

## Description of the Design

Use the "bundler" option, which has the features of Node16/NodeNext but doesn't force us yet to use file extensions:
> 'bundler' for use with bundlers. Like node16 and nodenext, this mode supports package.json "imports" and "exports", but unlike the Node.js resolution modes, bundler never requires file extensions on relative paths in imports.

### PR DESCRIPTION
**Pull request by turadg**
closes: #9184

## Description

In the course of https://github.com/Agoric/agoric-sdk/pull/9111 I've run continually run into problems with module resolution. Endo expects ESM but agoric-sdk is configured with a mixture of ESM and CommonJS.

This PR changes the tsconfig to `bundler` so that we can use export maps from packages. Then it has a slew of changes to accomodate that.

### Security Considerations

n/a
### Scaling Considerations

n/a

### Documentation Considerations

Perhaps note the `globalThis` type

### Testing Considerations

CI

### Upgrade Considerations

Should not have a chain impact

### CODE REVIEW

File: packages/ERTP/exported.d.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ *   to be ambient
+ */
+
+// export everything
+export * from './src/types.js';
+
+// XXX re-export types into global namespace, for consumers that expect these to
+//  be ambient
+import {
+  Amount as _Amount,
```
Comment (michaelfig) [inline packages/ERTP/exported.d.ts:15]: Can you explain somewhere (if you haven't already) why these need to be
```ts
import {
  Amount as _Amount,
} from './src/types.js';
declare global {
  export {
    _Amount as Amount,
  };
};
```

and not:
```ts
import {
  Amount,
} from './src/types.js';
declare global {
  export {
    Amount,
  };
};
```
?

File: packages/ERTP/src/issuerKit.js [hunk 2]
```diff
@@ -9,7 +9,8 @@ import { AssetKind, assertAssetKind } from './amountMath.js';
 import { coerceDisplayInfo } from './displayInfo.js';
 import { preparePaymentLedger } from './paymentLedger.js';
 
-import './types-ambient.js';
+/** @import {AdditionalDisplayInfo, RecoverySetsOption, IssuerKit, PaymentLedger} from './types.js' */
+/** @import {ShutdownWithFailure} from '@agoric/swingset-vat') */
```
Comment (michaelfig) [inline packages/ERTP/src/issuerKit.js]: Stray paren:
```suggestion
/** @import {ShutdownWithFailure} from '@agoric/swingset-vat' */
```

File: packages/ERTP/src/paymentLedger.js [hunk 3]
```diff
@@ -10,6 +10,11 @@ import { preparePurseKind } from './purse.js';
 import '@agoric/store/exported.js';
 import { BrandI, makeIssuerInterfaces } from './typeGuards.js';
 
+/**
+ * @import {Amount, AssetKind, DisplayInfo, PaymentLedger, Payment, Brand, RecoverySetsOption, Purse, Issuer, Mint} from './types.js'
+ * @import {ShutdownWithFailure} from '@agoric/swingset-vat')
```
Comment (michaelfig) [inline packages/ERTP/src/paymentLedger.js]: Again:
```suggestion
 * @import {ShutdownWithFailure} from '@agoric/swingset-vat'
```

File: packages/ERTP/src/types.js
Comment (michaelfig) [inline packages/ERTP/src/types.js:1]: Renaming this and making it non-ambient is great!  BTW, you'll have to clean up the `packageJson.scripts.prepack` and `packageJson.scripts.postpack` in order for packing to pass.  And also in other packages with the same `types-ambient.js` rename.

**Review by michaelfig — COMMENTED** [review]
No surprises here, just a whole lotta housekeeping!

File: packages/ERTP/src/types.js
Comment (michaelfig) [reply packages/ERTP/src/types.js:1]: Saw you resolved this already.

File: packages/agoric-cli/src/commands/inter.js [hunk 4]
```diff
@@ -138,7 +138,7 @@ const coerceBid = (offerStatus, agoricNames, warn) => {
  *
  * @param {import('@agoric/smart-wallet/src/offers.js').OfferStatus &
  *         { offerArgs: BidSpec}} bid
- * @param {import('agoric/src/lib/format.js').AssetDescriptor[]} assets
```
Comment (kriskowal) [inline packages/agoric-cli/src/commands/inter.js:141]: Did the reflexive import not work?

**Review by kriskowal — APPROVED** [review]

File: packages/agoric-cli/src/commands/inter.js (hunk 4 above)
Comment (turadg) [reply packages/agoric-cli/src/commands/inter.js:141]: didn't try it, was in a manual mechanical mode

**Pull request merged**